### PR TITLE
step description update xcode 13 test repetition

### PR DESCRIPTION
**Pull request by BanyikAnna**
### Checklist

- [ ] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MAJOR / MINOR / PATCH_ [version update](https://semver.org/)

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -254,7 +254,9 @@ inputs:
       category: Debug
       title: "Should retry test on failure?"
       description: |-
-        If `should_retry_test_on_fail: yes` step will retry the test if first attempt failed.
+        If you set this input to `yes`, the Step will rerun only the failed test case without having to rerun all the test cases. 
+        Please note that this feature is only available from Xcode 13 and above, earlier versions support rerunning ALL test cases instead of a specific failed one. 
+        You can use this new Xcode feature with the **Xcode Step for iOS** Step from Step version 2.7.x and above.
```
Comment (godrei) [inline step.yml:259]: I think this line is not needed: from the next version of the Step this feature will be available, in case of earlier version users will not see this line.

Comment (BanyikAnna) [reply step.yml:259]: Thank you @godrei :)

**Review by lpusok — APPROVED** [review]

**Review by godrei — DISMISSED** [review]

File: step.yml [hunk 2]
```diff
@@ -261,7 +261,8 @@ inputs:
       category: Debug
       title: "Should retry test on failure?"
       description: |-
-        If `should_retry_test_on_fail: yes` step will retry the test if first attempt failed.
+        If you set this input to `yes`, the Step will rerun only the failed test case without having to rerun all the test cases. 
+        Please note that this feature is only available from Xcode 13 and above, earlier versions support rerunning ALL test cases instead of a specific failed one.
```
Comment (Bence1001) [inline step.yml:265]: I think this can be misleading for 2 reasons:
1. "[...] the Step will rerun only the failed test case without having to rerun all the test cases." -> This is true, but not in this context. The step will rerun only the failed tests because it runs on stacks that support Xcode 13 or above, not because this switch is turned on.
2. We don't have another switch nor a "default behavior" that reruns all tests on failure.

I'd suggest rewording it, so that it articulates that:
1. This option allows you to retry your failed tests.
2. With Xcode 13 and above, it will retry only your failed tests. Below that, it will rerun all of your tests.

Wdyt? Does that make sense?

File: step.yml [hunk 3]
```diff
@@ -261,7 +261,8 @@ inputs:
       category: Debug
       title: "Should retry test on failure?"
       description: |-
-        If `should_retry_test_on_fail: yes` step will retry the test if first attempt failed.
+        If you set this input to `yes`, the Step will rerun only the failed test case without having to rerun all the test cases.
```
Comment (zoltan-baba) [inline step.yml:264]:
```suggestion
       If you set this input to `yes`, the Step will rerun your tests. With Xcode 13 and above, only your failed test cases will be rerun. With older Xcode versions, all test cases will be rerun. 
```

File: step.yml (hunk 2 above)
Comment (zoltan-baba) [reply step.yml:265]:
```suggestion
```

Comment (zoltan-baba) [reply step.yml:265]: I've added my suggestion above; the "please note" part can be deleted.

File: step.yml (hunk 3 above)
Comment (Bence1001) [reply step.yml:264]: LGTM! 👍 

@godrei @lpusok Wdyt?

**Review by Bence1001 — APPROVED** [review]
